gensim/models/word2vec.py

```diff
@@ -1116,17 +1116,36 @@ def doesnt_match(self, words):
         dists = dot(vectors, mean)
         return sorted(zip(dists, words))[0][1]
 
-    def __getitem__(self, word):
+    @classmethod
+    def sum(self, vecs):
+        '''
+        To compute the mean of N vectors
+        Example:
+        >>> sum(trained_model['office','products']
+          array([ -1.40128313e-02, ...]
+
+        '''
+        if len(vecs) > 1:
+            return matutils.unitvec(array(vecs).mean(axis=0)).astype(REAL)
+        else:
+            return vecs[0]
+
+    def __getitem__(self, words):
         """
-        Return a word's representations in vector space, as a 1D numpy array.
+        Return a list of N word representations in vector space, as a N D numpy array.
 
         Example::
 
-          >>> trained_model['woman']
+          >>> trained_model['office','products']
           array([ -1.40128313e-02, ...]
 
         """
-        return self.syn0[self.vocab[word].index]
+        if isinstance(words, string_types):
+            # allow calls like trained_model['office'], as a shorthand for trained_model[['dog']]
+            return self.syn0[self.vocab[words].index]
```

Comment:
It would be clean if we did `words = [words]` BUT it would result in returning a 2D array in all cases then. This `trianed_model["x"] = [[1.22323, ]]`, would be the result then. Which is against the default `__getitem__()`'s functionality. We need to have it the above way to make it return 1D and 2D arrays depending on the input or else it would end up in returning 2D arrays always.